src/components/fichas/Ficha.jsx

```diff
@@ -1,65 +1,51 @@
+import gatilloCaba from "../../data/gatilloCaba.json";
+import reportesCaba from "../../data/reportesCaba.json";
 
-const Ficha = () => {
-  const { Contador } = useParams();
-  const [caso, setCaso] = useState(null);
-  let tipoCaso;
-
-  // Determinar el tipo de caso según el primer carácter del Contador
-  if (Contador[0] === "d") {
-    tipoCaso = "dependencias";
-  } else if (Contador[0] === "g") {
-    tipoCaso = "gatillo";
-  } else if (Contador[0] === "r") {
-    tipoCaso = "reportes";
+// Determinar el tipo de caso según el primer carácter del Contador
+const getTipoDeCasoByContadorParam = (contador) => {
+  if (contador[0] === "d") {
+    return "dependencias";
+  } else if (contador[0] === "g") {
+    return "gatillo";
+  } else if (contador[0] === "r") {
+    return "reportes";
   } else {
-    tipoCaso = "no encontrado";
+    return "no encontrado";
   }
+};
 
-  useEffect(() => {
-    const fetchCaso = async () => {
-      try {
-        // Obtener los datos del archivo JSON correspondiente al tipo de caso
-        const response = await fetch(`data/${tipoCaso}Caba.json`);
-        const data = await response.json();
-
-        // Encontrar el caso por Contador
-        const casoEncontrado = data.features.find(
-          (c) => c.properties.Contador === Contador,
-        );
-        if (casoEncontrado) {
-          setCaso(casoEncontrado);
-        } else {
-          console.log("Caso no encontrado");
-        }
-      } catch (error) {
-        console.error("Error al cargar los datos del caso:", error);
-      }
-    };
-
-    fetchCaso();
-  }, [Contador, tipoCaso]);
+const titleByTipoCaso = {
+  gatillo: "Caso de gatillo fácil",
+  dependencias: "dependencia policial",
+  reportes: "Reporte de violencia policial",
+};
 
-  if (!caso) {
-    return <div>Cargando...</div>;
-  }
+const dataByTipoCaso = {
+  gatillo: gatilloCaba,
+  dependencias: dependenciasCaba,
+  reportes: reportesCaba,
+};
 
+const Ficha = () => {
+  const { Contador: contador } = useParams();
+  const tipoCaso = getTipoDeCasoByContadorParam(contador);
+  if (tipoCaso === "no encontrado") return <div>Tipo de caso inválido.</div>;
+  const dataDelTipoDelCaso = dataByTipoCaso[tipoCaso];
+  const caso = dataDelTipoDelCaso.features.find(
+    // FIXME Esto no está claro!
+    (c) => c.properties.Contador === contador,
+  );
+  if (!caso) return <div>Caso no encontrado.</div>;
   const { properties } = caso;
-
   return (
     <>
       <section className={styles.ficha}>
         <div className={styles.data}>
           <h2 className={styles.title}>
-            {tipoCaso === "gatillo"
-              ? "Caso de gatillo fácil"
-              : tipoCaso === "dependencias"
-                ? "dependencia policial"
-                : tipoCaso === "reportes"
-                  ? "Reporte de violencia policial"
-                  : "Tipo Desconocido"}
+            {titleByTipoCaso(tipoCaso) ?? "Tipo Desconocido"}
```

Comment:
Es un objeto, no una función!
TypeError: titleByTipoCaso is not a function

Reply:
sí, me saltó con TS. De hecho voy a mergear las dos ramas porque es un incordio si no

Reply:
oka, cierro entonce